plugins/router/f5/f5.go

```diff
@@ -0,0 +1,1743 @@
+	// the pool exists (i.e., the above GET request for the pool succeeds).
+	f5.poolMembers[poolname] = map[string]bool{}
+
+	for _, member := range res.Members {
+		f5.poolMembers[poolname][member.Name] = true
+	}
+
+	return f5.poolMembers[poolname], nil
+}
+
+// Check whether the specified pool exists.  Internally, this method uses
+// f5.poolMembers[poolname], as a side effect initialising it if it is zero.
+func (f5 *f5LTM) PoolExists(poolname string) (bool, error) {
+	_, err := f5.GetPoolMembers(poolname)
+	if err == nil {
+		return true, nil
+	}
+
+	if err.(F5Error).httpStatusCode == 404 {
+		return false, nil
+	}
+
+	return false, err
+}
+
+// Check whether the given member is in the pool named poolname on F5.
+// Internally, this method uses f5.poolMembers[poolname], initialising
+// it first if it is zero.
+func (f5 *f5LTM) PoolHasMember(poolname, member string) (bool, error) {
+	members, err := f5.GetPoolMembers(poolname)
+	if err != nil {
+		return false, err
+	}
+
+	return members[member], nil
+}
+
+// Add member to the pool named poolname on F5, and update
+// f5.poolMembers[poolname].
+func (f5 *f5LTM) AddPoolMember(poolname, member string) error {
+	hasMember, err := f5.PoolHasMember(poolname, member)
+	if err != nil {
+		return err
+	}
+	if hasMember {
+		glog.V(4).Infof("Pool %s already has member %s.\n", poolname, member)
+		return nil
+	}
+
+	glog.V(4).Infof("Adding pool member %s to pool %s.", member, poolname)
+
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s/members",
+		f5.host, poolname)
+
+	payload := struct {
+		Member string `json:"name"`
+	}{
+		Member: member,
+	}
+
+	err = f5.post(url, payload, nil)
+	if err != nil {
+		return err
+	}
+
+	members, err := f5.GetPoolMembers(poolname)
+	if err != nil {
+		return err
+	}
+
+	members[member] = true
+
+	glog.V(4).Infof("Added pool member %s to pool %s.",
+		member, poolname)
+
+	return nil
+}
+
+// Delete member from the pool named poolname on F5, and update
+// f5.poolMembers[poolname].
+func (f5 *f5LTM) DeletePoolMember(poolname, member string) error {
+	// The invocation of f5.PoolHasMember has the side effect that it will
+	// initialise f5.poolMembers[poolname], which is used below, if necessary.
+	hasMember, err := f5.PoolHasMember(poolname, member)
+	if err != nil {
+		return err
+	}
+	if !hasMember {
+		glog.V(4).Infof("Pool %s does not have member %s.\n", poolname, member)
+		return nil
+	}
+
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s/members/%s",
+		f5.host, poolname, member)
+
+	err = f5.delete(url, nil)
+	if err != nil {
+		return err
+	}
+
+	delete(f5.poolMembers[poolname], member)
+
+	glog.V(4).Infof("Pool member %s deleted from pool %s.", member, poolname)
+
+	return nil
+}
+
+// Return f5.routes[policyname], first initializing it from F5 if it is zero.
+func (f5 *f5LTM) getRoutes(policyname string) (map[string]bool, error) {
+	routes, ok := f5.routes[policyname]
+	if ok {
+		return routes, nil
+	}
```

Comment:
Similar comment to above - local cache - should flush this out at some interval and something for a follow up PR maybe.